core/src/main/java/com/linecorp/armeria/internal/PathMappingUtil.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright 2018 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.internal;
+
+import java.util.Map;
+import java.util.regex.Pattern;
+
+import com.linecorp.armeria.server.PathMapping;
+
+public final class PathMappingUtil {
+
+    //FIXME: Duplicate of {@link DefaultPathMapping#VALID_PATTERN}
+    private static final Pattern VALID_DEFAULT_PATTERN = Pattern
+            .compile("(?:/[^/{}:]+|/:[^/{}]+|/\\{[^/{}]+})+/?");
+
+    //FIXME: Currently unused
+    public static final String EXACT_PREFIX = "exact:";
+    public static final String PREFIX_PREFIX = "prefix:";
+    public static final String GLOB_PREFIX = "glob:";
+    public static final String REGEX_PREFIX = "regex:";
+
+    private PathMappingUtil() {}
+
+    public static String populateParams(PathMapping pathMapping, Map<String, String> pathParams) {
+        String newPath = pathMapping.exactPath().orElse(pathMapping.toString());
+        if (!isDefaultPath(newPath)) {
+            throw new IllegalArgumentException("pathMapping: " + pathMapping +
+                    " (expected: DefaultPathMapping)");
+        }
+        //find out if old path and new path are compatible
+        for (String param : pathMapping.paramNames()) {
+            if (!pathParams.containsKey(param)) {
+                throw new IllegalArgumentException("pathParams: " + param + " (no matching param in " +
+                        pathParams.toString() + ")");
+            }
```

Comment:
All the verification code in this function doesn't have to be performed on every redirection. We could move most of them to the constructor of `RedirectService`.